operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/kubernetes/DesiredResourceSanitizer.java

```diff
@@ -0,0 +1,47 @@
+package io.javaoperatorsdk.operator.processing.dependent.kubernetes;
+
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.kubernetes.api.model.PersistentVolumeClaimStatus;
+import io.fabric8.kubernetes.api.model.Secret;
+import io.fabric8.kubernetes.api.model.apps.StatefulSet;
+import io.javaoperatorsdk.operator.api.reconciler.Context;
+
+public class DesiredResourceSanitizer {
+
+  private DesiredResourceSanitizer() {}
+
+  public static <R, P extends HasMetadata> void sanitizeDesired(R desired, R actual, P primary,
+      Context<P> context, boolean useSSA) {
+    if (useSSA) {
+      if (desired instanceof StatefulSet) {
+        fillDefaultsOnVolumeClaimTemplate((StatefulSet) desired);
+      }
+      if (desired instanceof Secret) {
+        checkIfStringDataUsed((Secret) desired);
+      }
+    }
+  }
+
+  private static void checkIfStringDataUsed(Secret secret) {
+    if (secret.getStringData() != null && !secret.getStringData().isEmpty()) {
+      throw new IllegalStateException(
+          "There is a known issue using StringData with SSA. Use data instead.");
+    }
+  }
+
+  private static void fillDefaultsOnVolumeClaimTemplate(StatefulSet statefulSet) {
+    if (!statefulSet.getSpec().getVolumeClaimTemplates().isEmpty()) {
```

Comment:
Where are these values coming from? Isn't that what the default mutator webhook does? This seems like a slippery slope because it seems that we basically need to replicate what the admission controller is doing, no?

Reply:
There are no dynamic admission controllers registered by default in k8s, only if someone add one.

Reply:
Regardless, that basically mimics whatever Kubernetes is doing by default, meaning that this will break if they change…

Reply:
But I guess we can cross that bridge if/when we get there

Reply:
yep, also this is easy to opt out.